WordPress/src/main/java/org/wordpress/android/ui/posts/PostListDialogHelper.kt

```diff
@@ -28,7 +29,8 @@ private const val POST_TYPE = "post_type"
 class PostListDialogHelper(
     private val showDialog: (DialogHolder) -> Unit,
     private val checkNetworkConnection: () -> Boolean,
-    private val analyticsTracker: AnalyticsTrackerWrapper
+    private val analyticsTracker: AnalyticsTrackerWrapper,
+    private val showConflictResolutionOverlay: ((PostModel) -> Unit)? = null
```

Comment:
Just asking here cause I am not sure and wondering. Do we really need to add `showConflictResolutionOverlay` to the PostListDialogHelper? Why not just show the overlay directly from the PostActionHandler instead of calling  `postListDialogHelper.showConflictedPostResolutionDialog(post)`?

 ```
private fun performChecksAndEdit(site: SiteModel, post: PostModel) {
        // first of all, check whether this post is in Conflicted state with a more recent remote version
        if (doesPostHaveUnhandledConflict.invoke(post)) {
            postListDialogHelper.showConflictedPostResolutionDialog(post)
            return
        }

        // Then check if an autosave revision is available
        if (hasUnhandledAutoSave.invoke(post)) {
            postListDialogHelper.showAutoSaveRevisionDialog(post)
            return
        }

        editPost(site, post)
    }
```
Please ignore this question if I am missing something or if you think that this will complicate things instead of simplify them.

Reply:
I added it here to keep all the logic tidy and together; especially since the button taps logic is also in the helper. I didn't want to introduce new code that copy/pasted from existing. Let me see how the action logic plays out first.